src/groups/mqb/mqbc/mqbc_clusternodesession.h

```diff
@@ -392,7 +392,18 @@ ClusterNodeSession::onDispatcherEvent(const mqbi::DispatcherEvent& event)
     // the only events expected here are PUSH and ACK, for now.
 
     mqbi::DispatcherEvent& ev = const_cast<mqbi::DispatcherEvent&>(event);
-    ev.setClusterNode(d_clusterNode_p);
+    // TODO refactor
+    if (mqbi::DispatcherEventType::e_PUSH == ev.type()) {
+        const_cast<mqbi::DispatcherPushEvent*>(ev.asPushEvent())
+            ->setClusterNode(d_clusterNode_p);
```

Comment:
Is the refactoring here to make `asPushEvent` and family to have a const and non-const overload?